src/api/queries/useGetGoogleCalendar.tsx

```diff
@@ -1,30 +1,38 @@
 import { Session } from '@supabase/supabase-js';
 import { useQuery } from '@tanstack/react-query';
 
+import { formatEvents } from '../../components/Calendar/hooks/events';
+import { CALENDAR_TIME_MIN } from '../../constants/constants';
+import { QUERY_KEYS } from '../../constants/query_keys';
+import { createGoogleCalendarClient } from '../axios_instances/googleCalendarClient';
 import { GoogleCalendarEventsList } from '../types/googleCalendarEventsTypes';
 
 const getEvents = async (session: Session | null) => {
-  const TIME_MIN = '2022-01-01T00:00:00Z';
-  const url = new URL('https://www.googleapis.com/calendar/v3/calendars/primary/events');
-  url.searchParams.append('timeMin', TIME_MIN);
-
+ 
   if (session) {
-    const res = await fetch(url, {
-      method: 'GET',
-      headers: {
-        Authorization: 'Bearer ' + session.provider_token,
+  const googleCalendarClient=createGoogleCalendarClient(session)
+  const {data} = await googleCalendarClient.get("", {
+      params: {
+        'timeMin': CALENDAR_TIME_MIN,
       },
     });
-    const data = await res.json();
+    
     return data;
   }
 };
 
-export const useGetGoogleCalendarEvents = (session: Session | null) => {
+export const useGetGoogleCalendarEvents = (session: Session|null) => {
   const { data, error, isLoading } = useQuery<GoogleCalendarEventsList>({
-    queryKey: ['GET_EVENTS'],
+    queryKey: [QUERY_KEYS.getEvents],
     queryFn: () => getEvents(session),
     enabled: !!session,
   });
+  if(data){
+    return {
+      data: formatEvents(data),
+      error,
+      isLoading
+    }
+  }
   return { data, isLoading, error };
```

Comment:
w sumie to tutaj może jeszcze zrobić data: null , żeby nigdy nie pokazały się dane które nie są przeformatowane